### DISCUSSION TITLE
Does `cargo release [level]` actually run all the steps?

### DISCUSSION THREAD

**Discussion by linrongbin16** [crate-ci/cargo-release#832] [Q&A] [answered]
hi, in the [reference](https://github.com/crate-ci/cargo-release/blob/master/docs/reference.md), I found there are multiple steps: `version`, `changes`, etc.

But when I usually use this command, I simply run `cargo release major/minor`.

My question is: does the simple command actually runs all those steps (mentioned above) one by one in order?

If I want to deeply customize the release workflow, so I can customize these steps in a shell script?

For example in a cargo workspace, I always use the same version for all crates. So I still want to release/publish all crates to the crates.io registry, but with 1 single git tag pushed to github.

**Comment by epage** [comment, marked as answer]
`cargo release <level>` is a culmination of each step, except `cargo release changes`.  We offer the steps for error recovery and greater customization.

> For example in a cargo workspace, I always use the same version for all crates. So I still want to release/publish all crates to the crates.io registry, but with 1 single git tag pushed to github.

You can use `workspace.package.version` for this or put these packages in the same "version group".  You can also disable tagging for all packages and then re-enable it for the one package you want the tag created for.  The steps aren't needed for this.

**Reply by linrongbin16** [reply]
Thanks!
